### DISCUSSION TITLE
Is there an impact of frequent terminate workflow?

### DISCUSSION THREAD

**Discussion by DiyuanZhang** [Netflix/conductor#3709] [Q&A]
## Workflow Definition
My workflow is like **task A -> task B -> task C -> task D -> task E -> ...**

## Requirement
The requirement is when **task D** is **IN_PROGRESS**. The user can select to move the workflow back to **task B**.

## Current Solution
My current solution is when the user selects **move back**. Our system will call the **Terminate Workflow API** first and then the **Rerun Completed Workflow API**(with the **reRunFromTaskId** being task B).

## Question
I'm not sure whether there is a good solution. But it seems like there is no other way to implement the function like **Rerun From Specific Task** or **Move Back Workflow**. 

Since this solution maybe terminate workflow multiple times, so is there an impact of frequent terminate workflow?

**Comment by v1r3n** [comment]
Hi @DiyuanZhang This is a feature we are actively working on.  Will keep this thread updated as it gets available.

**Reply by DiyuanZhang** [reply]
Thanks @v1r3n 
I also submitted a feature request [[FEATURE]: Support GOTO/move back the previous Task](https://github.com/Netflix/conductor/issues/3710).
